File: sdks/php/src/Model/Credential.php
<?php

declare(strict_types=1);

namespace Atap\Sdk\Model;

/**
 * A W3C Verifiable Credential.
 */
final class Credential
{
    public function __construct(
        public readonly string $id = '',
        public readonly string $type = '',
        public readonly string $credential = '',
        public readonly string $issuedAt = '',
        public readonly ?string $revokedAt = null,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            type: (string) ($data['type'] ?? ''),
            credential: (string) ($data['credential'] ?? ''),
            issuedAt: (string) ($data['issued_at'] ?? ''),
            revokedAt: $data['revoked_at'] ?? null,
        );
    }
}

File: sdks/java/src/main/java/dev/atap/sdk/model/Credential.java
package dev.atap.sdk.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonProperty;

/**
 * A W3C Verifiable Credential.
 */
@JsonIgnoreProperties(ignoreUnknown = true)
public class Credential {

    @JsonProperty("id")
    private String id;

    @JsonProperty("type")
    private String type;

    @JsonProperty("credential")
    private String credential;

    @JsonProperty("issued_at")
    private String issuedAt;

    @JsonProperty("revoked_at")
    private String revokedAt;

    public Credential() {
    }

    public String getId() { return id; }
    public void setId(String id) { this.id = id; }

    public String getType() { return type; }
    public void setType(String type) { this.type = type; }

    public String getCredential() { return credential; }
    public void setCredential(String credential) { this.credential = credential; }

    public String getIssuedAt() { return issuedAt; }
    public void setIssuedAt(String issuedAt) { this.issuedAt = issuedAt; }

    public String getRevokedAt() { return revokedAt; }
    public void setRevokedAt(String revokedAt) { this.revokedAt = revokedAt; }
}
